src/ClusterManager/db_manager.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import logging.config
+import os
+import sys
+import time
+import yaml
+
+
+from cluster_manager import setup_exporter_thread, \
+    manager_iteration_histogram, \
+    register_stack_trace_dump, \
+    update_file_modification_time
+
+sys.path.append(os.path.join(os.path.dirname(
+    os.path.abspath(__file__)), "../utils"))
+
+from DataHandler import DataHandler
+from config import config
+
+
+DAYS_AGO = 1
+logger = logging.getLogger(__name__)
+
+
+def create_log(logdir='/var/log/dlworkspace'):
+    if not os.path.exists(logdir):
+        os.system("mkdir -p " + logdir)
+
+    with open('logging.yaml') as f:
+        logging_config = yaml.full_load(f)
+
+    log_filename = os.path.join(logdir, "db_manager.log")
+    logging_config["handlers"]["file"]["filename"] = log_filename
+    logging.config.dictConfig(logging_config)
+
+
+def delete_old_cluster_status():
+    table = "clusterstatus"
+    with DataHandler() as data_handler:
+        num_rows = data_handler.count_rows(table)
+        if num_rows <= 10:  # Retain 10 rows for safety
+            return
+
+        logger.info("Deleting rows from table %s older than %s day(s)", table,
+                    DAYS_AGO)
+        ret = data_handler.delete_rows_from_table_older_than_days(
+            table, DAYS_AGO)
+        ret_status = "succeeded" if ret is True else "failed"
+        logger.info("Deleting rows from table %s older than %s day(s) %s",
+                    table, DAYS_AGO, ret_status)
+
+
+def run():
+    register_stack_trace_dump()
+    create_log()
+    while True:
+        update_file_modification_time("db_manager")
+
+        with manager_iteration_histogram.labels("db_manager").time():
+            try:
+                delete_old_cluster_status()
+            except:
+                logger.exception("Deleting old cluster status failed",
+                                 exc_info=True)
+        time.sleep(600)

```

Comment:
This can be a daily job, say sleep 86400. Alternative is to create a new cluster garbage collection service, a k8s cron job. -- to clean DB as well as check the orphan pods in k8s.

Reply:
Let's use 86400 here and create a cron job service in a later PR.